feature/bgp/gracefulrestart/otg_tests/bgp_graceful_restart_test/bgp_graceful_restart_test.go

```diff
@@ -0,0 +1,677 @@
+	t.Run("verifyDUTPorts", func(t *testing.T) {
+		t.Log("Verifying port status")
+		verifyPortsUp(t, dut.Device)
+	})
+	t.Run("VerifyBGPParameters", func(t *testing.T) {
+		t.Log("Check BGP parameters")
+		checkBgpStatus(t, dut)
+	})
+
+	var startTime time.Time
+	// Send Graceful Restart Trigger from ATE to DUT within the GR timer configured on the DUT
+	t.Run("InitiateGracefulRestartWithinDUTGRTimer", func(t *testing.T) {
+		// starting traffic before graceful restart
+		t.Log("Starting traffic before Graceful restrat Trigger from OTG")
+		ate.OTG().StartTraffic(t)
+		startTime = time.Now()
+		t.Log("Send Traffic while GR timer counting down. Traffic should pass as BGP GR is enabled!")
+		t.Log("Send Graceful Restart Trigger from OTG to DUT -- trigger GR Restart timer(60s) < configured GR restart Timer(120s)")
+		ate.OTG().SetControlAction(t, createGracefulRestartAction(t, []string{"atedst.BGP4.peer"}, 60))
+	})
+	statePath := gnmi.OC().NetworkInstance(deviations.DefaultNetworkInstance(dut)).Protocol(oc.PolicyTypes_INSTALL_PROTOCOL_TYPE_BGP, "BGP").Bgp()
+	nbrPath := statePath.Neighbor(ateDst.IPv4)
+
+	t.Run("VerifyBGPNOTEstablished", func(t *testing.T) {
+		t.Log("Waiting for BGP neighbor to go to ACTIVE state after GR Trigger to DUT...")
+		bgpState := oc.Bgp_Neighbor_SessionState_ACTIVE
+		_, ok := gnmi.Watch(t, dut, nbrPath.SessionState().State(), 2*time.Minute, func(val *ygnmi.Value[oc.E_Bgp_Neighbor_SessionState]) bool {
+			currState, ok := val.Val()
+			return ok && currState == bgpState
+		}).Await(t)
+		if !ok {
+			fptest.LogQuery(t, "BGP reported state", nbrPath.State(), gnmi.Get(t, dut, nbrPath.State()))
+			t.Errorf("BGP session did not go Down as expected")
+		}
+	})
+
+	const stopDuration = 45 * time.Second
+	replaceDuration := time.Since(startTime)
+	time.Sleep(grTimer - stopDuration - replaceDuration)
+	ate.OTG().StopTraffic(t)
+	t.Log("Traffic stopped")
+	verifyNoPacketLoss(t, ate, config)
+
+	t.Run("VerifyBGPEstablished", func(t *testing.T) {
+		t.Logf("Waiting for BGP neighbor to establish...")
+		_, ok := gnmi.Watch(t, dut, nbrPath.SessionState().State(), 2*time.Minute, func(val *ygnmi.Value[oc.E_Bgp_Neighbor_SessionState]) bool {
+			currState, ok := val.Val()
+			return ok && currState == oc.Bgp_Neighbor_SessionState_ESTABLISHED
+		}).Await(t)
+		if !ok {
+			fptest.LogQuery(t, "BGP reported state", nbrPath.State(), gnmi.Get(t, dut, nbrPath.State()))
+			t.Errorf("BGP session not Established as expected")
+		}
+	})
+
+	// Send Graceful Restart Trigger from ATE to DUT after the expiration of the GR timer configured on the DUT
+	t.Run("InitiateGracefulRestartExceedingDUTGRTimer", func(t *testing.T) {
+		t.Log("Send Graceful Restart Trigger from OTG to DUT -- Trigger GR Restart timer(140s) > configured GR Stale Time (120s) ")
+		ate.OTG().SetControlAction(t, createGracefulRestartAction(t, []string{"atedst.BGP4.peer"}, grStaleRouteTime+20))
+		time.Sleep(time.Second * grStaleRouteTime)
+		ate.OTG().StartTraffic(t)
+	})
```

Comment:
I don't think we're distinguishing between the GR restart/stale times in this case. (As I understand it, the routes would be dropped from the RIB after the advertised restart time from the ATE anyways.)

@sachendras :
- Is my understanding correct here?
- Is it possible (if arbitrary) to test the stale time setting on the DUT by setting it lower than the GR restart time? Or is that invalid configuration?